src/rust/en.reaperscans/src/parser.rs

```diff
@@ -0,0 +1,413 @@
+	let has_more = pagination.contains("Next");
+
+	Ok(MangaPageResult {
+		manga: mangas,
+		has_more,
+	})
+}
+
+pub fn parse_manga_details(base_url: String, manga_id: String) -> Result<Manga> {
+	let url = get_manga_url(manga_id.clone(), base_url);
+
+	let html = Request::new(&url, HttpMethod::Get)
+		.header("User-Agent", &get_user_agent())
+		.html()
+		.expect("Reaperscans: Failed to access the comic page. Try accessing the website on a browser and check your internet connection");
+
+	let cover = html
+		.select("main div.grid div.container img")
+		.attr("src")
+		.read();
+	let title = String::from(
+		html.select("main div.grid div.container h1")
+			.text()
+			.read()
+			.trim(),
+	);
+	let description = text_with_newlines(html.select("main div.grid section div > div.p-4 > p"));
+
+	let mut status = MangaStatus::Unknown;
+	let mut age_rating = String::new();
+	let mut nsfw = MangaContentRating::Safe;
+	let mut language = String::new();
+
+	for node in html
+		.select("main div.grid section div > div.p-4 > div > dl > div")
+		.array()
+	{
+		let info = node
+			.as_node()
+			.expect("Reaperscans: Could not parse the comic's details")
+			.text()
+			.read();
+
+		if info.contains("Source Language") {
+			language = String::from(info.replace("Source Language", "").trim());
+		} else if info.contains("Age Rating") {
+			age_rating = info.replace("Age Rating", "");
+		} else if info.contains("Release Status") {
+			let release_status = info.replace("Release Status", "");
+
+			if release_status.contains("Ongoing") {
+				status = MangaStatus::Ongoing;
+			} else if release_status.contains("Complete") {
+				status = MangaStatus::Completed;
+			} else if release_status.contains("Dropped") {
+				status = MangaStatus::Cancelled;
+			} else if release_status.contains("On hold") {
+				status = MangaStatus::Hiatus;
+			}
+		}
+	}
+
+	if age_rating.contains("18+") {
+		nsfw = MangaContentRating::Nsfw;
+	} else if age_rating.contains("16+") {
+		nsfw = MangaContentRating::Suggestive;
+	}
+
+	let viewer = match language.as_str() {
+		"Japanese" => MangaViewer::Rtl,
+		_ => MangaViewer::Scroll,
+	};
+
+	Ok(Manga {
+		id: manga_id,
+		cover,
+		title,
+		author: String::new(),
+		artist: String::new(),
+		description,
+		url,
+		categories: Vec::new(),
+		status,
+		nsfw,
+		viewer,
+	})
+}
+
+pub fn parse_chapter_list(base_url: String, manga_id: String) -> Result<Vec<Chapter>> {
+	let mut chapters: Vec<Chapter> = Vec::new();
+	let url = get_manga_url(manga_id, base_url.clone());
+	let request = Request::new(url, HttpMethod::Get).header("User-Agent", &get_user_agent());
+
+	let initial_request = request
+		.html()
+		.expect("Reaperscans: Could not retreive the chapter list html");
+
+	parse_chapter_list_helper(initial_request.clone(), &mut chapters);
+
+	let mut page = 2;
+	loop {
+		// If let here gracefully handles the case where cloudflare blocks the request.
+		// If the request is blocked, then the loop will continue and try again.
```

Comment:
could also be an infinite loop of network requests if the user is having network issues. it should have a max amount of tries, e.g. 5 or something, before it just returns an error instead of retrying